### PR TITLE
owner: fix global checkpointts stopping when it 's not exist changefees

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
If the number of changefeed is zero, checkpoint will not be pushing

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/changefeed.go [hunk 1]
```diff
@@ -481,7 +481,7 @@ func (c *changeFeed) calcResolvedTs(ctx context.Context) error {
 	minCheckpointTs := c.targetTs
 
 	if len(c.taskPositions) == 0 {
-		minCheckpointTs = c.status.CheckpointTs
+		minCheckpointTs = c.status.ResolvedTs
 	} else if len(c.taskPositions) < len(c.taskStatus) {
 		return nil
```
Comment (amyangfei) [inline cdc/changefeed.go:486]: Is this better:
- check if `len(c.taskPositions) < len(c.taskStatus)`, do nothing
- then check if `len(c.taskPositions) == 0`, update `minCheckpointTs` to `c.status.ResolvedTs`
If owner has dispatched a table but processor doesn't start replicating it, we should wait the processor is ready?

Comment (leoppro) [reply cdc/changefeed.go]: > If owner has dispatched a table but processor doesn't start replicating it, owner should wait until the processor is ready?

Yes, the owner should, and we using table lock to make the owner wait until the processor is ready now

**Comment by leoppro** [comment]
until the processor is ready

**Pull request merged**